rasa/nlu/extractors/mitie_entity_extractor.py

```diff
@@ -133,50 +185,114 @@ def _prepare_mitie_sample(training_example: Message) -> Any:
+
+        Args:
+            message: a user message
+            mitie_model: MitieModel containing a `mitie.total_word_feature_extractor`
 
-        ents = self.extract_entities(
-            message.get(TEXT), message.get(TOKENS_NAMES[TEXT]), mitie_feature_extractor
+        Returns:
+            a list of dictionaries describing the entities
+        """
+        text = message.get(TEXT)
+        tokens = message.get(TOKENS_NAMES[TEXT])
+
+        entities = []
+        token_texts = [token.text for token in tokens]
+        mitie_entities = self._ner.extract_entities(
+            token_texts, mitie_model.word_feature_extractor,
         )
-        extracted = self.add_extractor_name(ents)
-        message.set(ENTITIES, message.get(ENTITIES, []) + extracted, add_to_output=True)
+        for e in mitie_entities:
+            if len(e[0]):
+                start = tokens[e[0][0]].start
+                end = tokens[e[0][-1]].end
+
+                entities.append(
+                    {
+                        ENTITY_ATTRIBUTE_TYPE: e[1],
+                        ENTITY_ATTRIBUTE_VALUE: text[start:end],
+                        ENTITY_ATTRIBUTE_START: start,
+                        ENTITY_ATTRIBUTE_END: end,
+                        ENTITY_ATTRIBUTE_CONFIDENCE: None,
+                    }
+                )
+        return entities
 
     @classmethod
     def load(
         cls,
-        meta: Dict[Text, Any],
-        model_dir: Text,
-        model_metadata: Metadata = None,
-        cached_component: Optional["MitieEntityExtractor"] = None,
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
         **kwargs: Any,
-    ) -> "MitieEntityExtractor":
+    ) -> MitieEntityExtractorGraphComponent:
         """Loads trained component (see parent class for full docstring)."""
         import mitie
 
-        file_name = meta.get("file")
-
-        if not file_name:
-            return cls(meta)
-
-        classifier_file = os.path.join(model_dir, file_name)
-        if os.path.exists(classifier_file):
-            extractor = mitie.named_entity_extractor(classifier_file)
-            return cls(meta, extractor)
-        else:
-            return cls(meta)
+        try:
+            with model_storage.read_from(resource) as model_path:
+                ner_file = model_path / cls.MITIE_RESOURCE_FILE
+                if not ner_file.exists():
+                    raise FileNotFoundError(
+                        f"Expected a MITIE extractor file at {ner_file}."
+                    )
+                ner = mitie.named_entity_extractor(str(ner_file))
+                return MitieEntityExtractorGraphComponent(
+                    config, model_storage, resource, ner=ner
+                )
+        except (FileNotFoundError, Exception) as e:
+            if not isinstance(e, FileNotFoundError):
+                is_mitie_exception = not (
```

Comment:
Why do we need this? Can't we just have different `except` clauses?

Reply:
In both cases I want to create the default instance

Reply:
It just seems a bit brittle to condition on error message text from code we don't control. I'm also a bit confused, why is this called `is_mitie_exception` if there is the `not` it seems like that exception comes from mitie? Or am I missing something?

Reply:
oh yep that not was too much :) (had that statement in the if before) - yes it is brittle but it only catches things with that exact message (but I can also remove it if we don't expect any errors from mitie directly)

Reply:
another option is to have a separate `try except` just around `mitie.named_entity_extractor(str(ner_file)), which raises a specific warning. And then just handle `ValueError` in this one (similarly to how we do in other places we use `model_storage.read_from`)

Reply:
you mean raising a ValueError from the Exception (in case that comes from mitie and then handle that)? (just updated code accordingly)

Reply:
I was thinking this:

```python
        try:
            with model_storage.read_from(resource) as model_path:
                ner_file = model_path / cls.MITIE_RESOURCE_FILE
                if not ner_file.exists():
                    raise FileNotFoundError(
                        f"Expected a MITIE extractor file at {ner_file}."
                    )

                try:
                    ner = mitie.named_entity_extractor(str(ner_file))
                except Exception:
                    rasa.shared.utils.io.raise_warning(
                        f"Mitie wasn't able to load the MITIE extractor from "
                        f"{ner_file}."
                    )
                    return cls(config, model_storage, resource, ner=ner)

        except ValueError as e:
            rasa.shared.utils.io.raise_warning(
                f"Failed to load {cls.__name__} from model storage. "
                f"This can happen if the model could not be trained because regexes "
                f"could not be extracted from the given training data - and hence "
                f"could not be persisted. Error: {e}."
            )
            return cls(config, model_storage, resource)

```

Reply:
Wouldn't this also catch keyboard interrupts happening during loading of the mitie model?

Reply:
No, `KeyboardInterrupt` inherits from `BaseException` but not `Exception`.

Reply:
tbh though I would be fine with not catching the mitie exception. Better to fail fast than suppress an error that shouldn't happen.

Reply:
I'll just remove it - we're not catching this stuff anywhere else I think. I was just triggered to look into it when you mentioned value error and then was wondering if that could come from there.

> inherits from BaseException

oh, nice that they made that... exception :D

Reply:
> oh, nice that they made that... exception :D

😂